icu4c/source/test/intltest/tstnorm.cpp

```diff
@@ -1607,12 +1608,12 @@ BasicNormalizerTest::TestNormalizeUTF8WithEdits() {
     assertTrue("isNormalizedUTF8(normalized)", nfkc_cf->isNormalizedUTF8(result, errorCode));
 
     // Omit unchanged text.
-    expected = reinterpret_cast<const char*>(u8"aääạ\u0308ạ\u0308가각갃");
+    expected = u8"aääạ\u0308ạ\u0308가각갃";
     result.clear();
     edits.reset();
     nfkc_cf->normalizeUTF8(U_OMIT_UNCHANGED_TEXT, src, sink, &edits, errorCode);
     assertSuccess("normalizeUTF8 omit unchanged", errorCode.get());
-    assertEquals("normalizeUTF8 omit unchanged", expected.c_str(), result.c_str());
+    assertEquals("normalizeUTF8 omit unchanged", expected.data(), result.c_str());
```

Comment:
You're not passing a `.data(), .size()` pair here, which I think means you're expecting both `expected` and `result` to be null-terminated C-style strings, which means `.c_str()` is semantically correct. For example, if you changed `expected` to be a `string_view` or `StringPiece`, this code would no longer work, because those types aren't logically guaranteed to be null-terminated.

If you want to use non-null-terminated string pieces here, you'll have to find a way to pass the number of chars to compare (i.e., pass both `.data` and `.size`, or pass a `string_view` or `StringPiece` that wraps up those two pieces of information into a single object).

Reply:
Right. StringPiece does not imply NUL-termination, but in this code I know what I am initializing the StringPieces with.